.github/workflows/main.yml

```diff
@@ -0,0 +1,94 @@
+name: Godot CI
+
+on:
+    pull_request:
+    push:
+      branches: [ master, 3.2, 3.1, 3.0 ]
+env:
+  SCONS_CACHE_MSVC_CONFIG: true
+  SCONS_CACHE_LIMIT: 2048
+jobs:
+  windows-editor:
+    runs-on: "windows-latest"
+    name: Windows Editor (release_debug)
+    steps:
```

Comment:
There's a lot of steps, do we need to be so fine-grained when the output we're most interested in will be: What python/scons env we have, and the actual Godot build logs and upload?

I'd suggest grouping all the Python setup under one named step.

Reply:
I smushed the python together so this should be better, but I would like to keep the python installer separate since it is official code from their plugin basically means updating this in the future if they change it is easier